lib/utils/config.ts

```diff
@@ -165,3 +176,55 @@ export async function validateDevOptionAndWarn(
 		and exposes network ports such as 2375 that allows unencrypted access to balenaEngine.
 		Therefore, development mode should only be used in private, trusted local networks.`);
 }
+
+/**
+ * Chech whether the `--secureBoot` option of commands related to OS configuration
+ * such as `os configure` and `config generate` is compatible with a given
+ * OS release, and print a warning regarding the consequences of using that
+ * option.
+ */
+export async function validateSecureBootOptionAndWarn(
+	secureBoot?: boolean,
+	slug?: string,
+	version?: string,
+	logger?: import('./logger'),
+) {
+	if (!secureBoot) {
+		return;
+	}
+	const { ExpectedError } = await import('../errors');
+	if (!version) {
+		throw new ExpectedError(`Error: No version provided`);
+	}
+	if (!slug) {
+		throw new ExpectedError(`Error: No device type provided`);
+	}
+	const sdk = getBalenaSdk();
+	const [OsRelease]: BalenaSdk.OsVersion[] =
+		await sdk.models.os.getAllOsVersions(`${slug}`, {
+			$select: 'id',
+			$filter: { raw_version: `${version.replace(/^v/, '')}` },
+		});
+	if (!OsRelease) {
+		throw new ExpectedError(`Error: No ${version} release for ${slug}`);
+	}
+	const OsContract = await sdk.models.release.get(OsRelease.id);
```

Comment:
You can actually combine the two requests, but atm the SDK typings will not include the added properties :(
Working on this in the next major, but for now it's better to cast it rather than doing 2 requests.
```suggestion
	// TODO: Remove the added type casting when the SDK returns the fully typed result
	const [osRelease] = (await sdk.models.os.getAllOsVersions(`${slug}`, releasePineOpts)) as Array<BalenaSdk.OsVersion & BalenaSdk.PineTypedResult<BalenaSdk.Release, typeof releasePineOpts>>;
	if (!osRelease) {
		throw new ExpectedError(`Error: No ${version} release for ${slug}`);
	}
	const osContract = osRelease.contract;
```
If you still prefer 2 requests, then we should at least add a $select in the second request, so that it avoids fetching unnecessary field (which in this resource can be quite large).

Reply:
thanks @thgreasi I'll try with one request as soon as I get back to this PR

Reply:
@thgreasi I wasn't able to do this in one request, and I don't quite follow the code sample typings unfortunately. I have left this in two requests with the additional select, but I also think one request is best. Do you know how the single request would look like?